backend/src/utils.ts

```diff
@@ -28,3 +29,18 @@ const htmlifyEditorNode = (node: NodeType): string => {
 export const htmlifyEditorNodes = (value: NodeType[]): string => {
   return value.map(htmlifyEditorNode).join('\n')
 }
+
+// Takes in an original Post or Comment and a currently logged in User and checks that
+// the currentUser has permission to update or delete that Post/Comment
+export const hasPostPermissions = (
+  original: Post | Comment,
+  currentUser: User,
+) => {
+  const hasPermission =
+    original.authorId == currentUser.id ||
+    currentUser.userRole !== 'MODERATOR' ||
+    'ADMIN'
```

Comment:
```suggestion
    currentUser.userRole !== 'ADMIN'
```

Think this will always return `true` as it stands.

That's cool that the same thing can be used for both posts and comments though